### PR TITLE
global_settings: better handling of exc in GuiVM

### PR DESCRIPTION
**Pull request by fepitre**

### CODE REVIEW

**Comment by fepitre** [comment]
PipelineRetry

**Comment by fepitre** [comment]
PipelineRetry

File: qubesmanager/global_settings.py [hunk 1]
```diff
@@ -50,16 +50,15 @@ def _run_qrexec_repo(service, arg=''):
...
     )
     if p.stderr:
-        raise exc.QubesException(
-            QtCore.QCoreApplication.translate(
-                "GlobalSettings", 'qrexec call stderr was not empty'),
-            {'stderr': p.stderr.decode('utf-8')})
+        msg = QtCore.QCoreApplication.translate(
+                "GlobalSettings",
+                'qrexec call stderr was not empty')
+        raise exc.QubesException(msg + '(%s)', p.stderr.decode('utf-8'))
```
Comment (marmarek) [inline qubesmanager/global_settings.py:56]:
```suggestion
        raise exc.QubesException(msg + ' (%s)', p.stderr.decode('utf-8'))
```

File: qubesmanager/global_settings.py [hunk 2]
```diff
@@ -50,16 +50,15 @@ def _run_qrexec_repo(service, arg=''):
...
     if p.returncode != 0:
-        raise exc.QubesException(
-            QtCore.QCoreApplication.translate(
+        msg = QtCore.QCoreApplication.translate(
                 "GlobalSettings",
-                'qrexec call exited with non-zero return code'),
-            {'returncode': p.returncode})
+                'qrexec call exited with non-zero return code')
+        raise exc.QubesException(msg + '(%s)', p.returncode)
```
Comment (marmarek) [inline qubesmanager/global_settings.py:61]:
```suggestion
        raise exc.QubesException(msg + ' (%s)', p.returncode)
```

**Review by marmarek — CHANGES_REQUESTED** [review]

**Review by marmarek — APPROVED** [review]

**Pull request merged**